include/ast.hpp

```diff
@@ -93,6 +93,18 @@ struct NullStmtNode : public StmtNode {
   virtual void Accept(ModifyingVisitor&) override;
 };
 
+struct IfStmtNode : public StmtNode {
+  IfStmtNode(std::unique_ptr<ExprNode> expr,
+             std::unique_ptr<BlockStmtNode> block)
+      : expr{std::move(expr)}, block{std::move(block)} {}
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+
+  std::unique_ptr<ExprNode> expr;
```

Comment:
```suggestion
  std::unique_ptr<ExprNode> predicate;
```
The expression should not only represent an expression but also have its type convertible to `bool` after evaluation.
I suggest that we change the name from `expr` to `predicate` to better reflect this characteristic.
Aslo `if_stmt->expr` can be a little bit confusing.